src/mixins/RoomSockets.ts

```diff
@@ -0,0 +1,43 @@
+import Vue from 'vue'
+import Component from 'vue-class-component'
+import { Socket } from 'vue-socket.io-extended'
+import { Action, Getter } from 'vuex-class'
+import { Prop, Watch } from 'vue-property-decorator'
+import { AuthenticationGetters, ExtendedJWT } from '@/store/modules/authentication'
+import { SocketActions } from '@/store/modules/socket'
+
+@Component({
+  name: 'RoomSocket'
+})
+export default class RoomSocket extends Vue {
+  @Prop() id!: string
+  @Action(`socket/${SocketActions.JOIN_ROOM}`) joinRoom!: (id: string) => void
+  @Getter(`authentication/${AuthenticationGetters.IS_AUTH}`) isAuth!: boolean
+  @Getter(`authentication/${AuthenticationGetters.JWT}`) jwt!: ExtendedJWT
+  @Socket()
+  connect () {
+    this.joinRoom(this.id)
+  }
+
+  @Socket('userJoined')
+  onUserJoined (jti: string) {
```

Comment:
When other people join the room, we get their JTI. We might need more than that, but that stuff is for later

Reply:
we concluded with using pascalCase, but with a namespace prefix